goalc/compiler/compilation/CompilerControl.cpp

```diff
@@ -616,3 +628,185 @@ Val* Compiler::compile_print_debug_compiler_stats(const goos::Object& form,
 
   return get_none();
 }
+
+Val* Compiler::compile_gen_docs(const goos::Object& form, const goos::Object& rest, Env*) {
+  auto args = get_va(form, rest);
+  va_check(form, args, {goos::ObjectType::STRING}, {});
+
+  const auto& doc_path = fs::path(args.unnamed.at(0).as_string()->data);
+  lg::info("Saving docs to: {}", doc_path.string());
+
+  const auto symbols = m_symbol_info.get_all_symbols();
+
+  std::unordered_map<std::string, Docs::SymbolDocumentation> all_symbols = {};
+  std::unordered_map<std::string, Docs::FileDocumentation> file_docs = {};
```

Comment:
minor nit: in C++ stuff like `std::vector`, string, map, optional, will all default being empty, so you don't need the ` = {}`.